packages/expo-sensors/android/src/main/java/expo/modules/sensors/modules/BaseSensorModule.kt

```diff
@@ -0,0 +1,83 @@
+package expo.modules.sensors.modules
+
+import android.content.Context
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener2
+import android.os.Bundle
+import android.util.Log
+import expo.modules.interfaces.sensors.SensorServiceInterface
+import expo.modules.interfaces.sensors.SensorServiceSubscriptionInterface
+import org.unimodules.core.ExportedModule
+import org.unimodules.core.ModuleRegistry
+import org.unimodules.core.interfaces.LifecycleEventListener
+import org.unimodules.core.interfaces.services.EventEmitter
+import org.unimodules.core.interfaces.services.UIManager
+
+abstract class BaseSensorModule internal constructor(context: Context?) : ExportedModule(context), SensorEventListener2, LifecycleEventListener {
+  lateinit var moduleRegistry: ModuleRegistry
+    private set
+  private val sensorKernelServiceSubscription: SensorServiceSubscriptionInterface by lazy {
+    getSensorService().createSubscriptionForListener(this)
+  }
+  private var mIsObserving = false
+
+  protected abstract val eventName: String
+  protected abstract fun getSensorService(): SensorServiceInterface
+  protected abstract fun eventToMap(sensorEvent: SensorEvent): Bundle
+
+  override fun onCreate(moduleRegistry: ModuleRegistry) {
+    // Unregister from old UIManager
+    if (this::moduleRegistry.isInitialized && moduleRegistry.getModule(UIManager::class.java) != null) {
+      moduleRegistry.getModule(UIManager::class.java).unregisterLifecycleEventListener(this)
+    }
+    this.moduleRegistry = moduleRegistry
```

Comment:
Let's clean it up a bit.
We can try to unregister the `lifecycleEventListener` even if it was not previously registered. These `if` conditions don't guard before anything to be honest 😅 
```suggestion
    this.moduleRegistry = moduleRegistry

    // Unregister from old UIManager
    moduleRegistry.getModule(UIManager::class.java)?.unregisterLifecycleEventListener(this)
```